packages/material-ui-lab/src/MonthPicker/PickersMonth.tsx

```diff
@@ -1,72 +1,102 @@
 import * as React from 'react';
 import clsx from 'clsx';
-import Typography from '@material-ui/core/Typography';
-import { MuiStyles, StyleRules, WithStyles, withStyles } from '@material-ui/core/styles';
+import Typography, { TypographyTypeMap } from '@material-ui/core/Typography';
+import {
+  experimentalStyled,
+  unstable_useThemeProps as useThemeProps,
+  Theme,
+} from '@material-ui/core/styles';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import { OverridableComponent } from '@material-ui/core/OverridableComponent';
 import { onSpaceOrEnter } from '../internal/pickers/utils';
+import pickersMonthClasses, { getPickersMonthUtilityClass } from './pickersMonthClasses';
 
 export interface MonthProps {
+  className?: string;
   children: React.ReactNode;
   disabled?: boolean;
   onSelect: (value: any) => void;
   selected?: boolean;
   value: any;
 }
 
-export type PickersMonthClassKey = 'root' | 'selected';
+export type PickersMonthClassKey = keyof typeof pickersMonthClasses;
 
-export const styles: MuiStyles<PickersMonthClassKey> = (
-  theme,
-): StyleRules<PickersMonthClassKey> => ({
-  root: {
-    flex: '1 0 33.33%',
-    display: 'flex',
-    alignItems: 'center',
-    justifyContent: 'center',
-    cursor: 'pointer',
-    outline: 0,
-    height: 64,
-    transition: theme.transitions.create('font-size', { duration: '100ms' }),
-    '&:focus': {
-      color: theme.palette.primary.main,
-      fontWeight: theme.typography.fontWeightMedium,
-    },
-    '&:disabled': {
-      pointerEvents: 'none',
+const useUtilityClasses = () => {
+  const slots = {
+    root: ['root'],
+  };
+
+  return composeClasses(slots, getPickersMonthUtilityClass, undefined);
+};
+
+const PickersMonthRoot = experimentalStyled<
+  OverridableComponent<TypographyTypeMap<{ component?: React.ElementType; disabled?: boolean }>>
+>(
+  Typography,
+  {},
+  {
+    name: 'MuiPickersMonth',
+    slot: 'Root',
+    overridesResolver: (props, styles) => styles.root,
+  },
+)(({ theme, styleProps = {} }) => ({
+  flex: '1 0 33.33%',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  cursor: 'pointer',
+  outline: 0,
+  height: 64,
+  transition: theme.transitions.create('font-size', { duration: '100ms' }),
+  '&:focus': {
+    color: theme.palette.primary.main,
+    fontWeight: theme.typography.fontWeightMedium,
+  },
+  '&:disabled': {
+    pointerEvents: 'none',
+    ...(!styleProps.selected && {
       color: theme.palette.text.secondary,
-    },
-    '&$selected': {
-      color: theme.palette.primary.main,
-      fontWeight: theme.typography.fontWeightMedium,
-    },
+    }),
```

Comment:
Need to add condition here due to styles overriding with emotion.
![image](https://user-images.githubusercontent.com/18292247/116681435-2e9c7380-a9d7-11eb-8668-d9ff6a83ef63.png)